src/v/cloud_storage/partition_manifest.h

```diff
@@ -702,6 +710,16 @@ class partition_manifest : public base_manifest {
     // insync offset is incremented after processing every batch (even if it's
     // skipped by the STM).
     model::offset _applied_offset;
+
+    /** the following fields are not serialized as of v24.1 **/
+
+    // Keeps track of how many bytes are compacted away in cloud_storage.
+    // This value is updated every time a new add(segment_meta) replaces some
+    // segments. It is not persisted because it's used only to observe cloud
+    // storage behavior. It is expected that this value does not decrease,
+    // except when recreating the manifest. a decreasing value would indicate a
+    // change in behavior (or a bug) in user of the add(segment_meta) operation.
+    size_t _compacted_away_cloud_bytes{0};
```

Comment:
Strawman idea: maybe it makes sense to track the size of removed segments that fell above the start offset? Does that capture the information we're trying to expose?

Reply:
If not, I think this is fine. Just something to think about.

Reply:
at first i tried to compute this value outside of the partition manifest, but i ran into two issues.
one is that it's difficult to determine retention/compaction. I think keeping track of "size above start offset" should help

the second one is that it's difficult to match a replaced segment A to the replacing segment B after the fact, and there is nothing stopping B to be replaced itself
(difficult also in the sense of not begin constant cost)

using partition_manifest is instead simple. 
since it's used in archival_metadata_stm, i could keep track of the value there and then ntp_archiver can fech the value when needed.

Reply:
Does the second problem exist if we're only tracking the number of bytes removed above the start offset? Wondering if there'd be something wrong with that approach (I'm open to it being not straightforward to implement and therefore sticking to the current approach, but just curious)

Reply:
sorry forgot to reply.
the second problem is born from the fact that the _replaced list is an unordered vector of segments and there is no direct link between a group of _replaced segments and the segment that replaced them, so no direct way to compute the difference in sizes. does this answer the question? anyway i moved the metric to a separate pr

Reply:
>there is no direct link between a group of _replaced segments and the segment that replaced them

In the case of this metric, though, why would we need to map them back to the original segments? Aren't we just incrementing the metric by the removed segment's size, provided the segment is above the start offset?

Anyway, looks like Evgeny has similar concerns about pushing some of this into the archiver. Will close out this discussion since this PR seems good to go